blockmanager.go

```diff
@@ -309,10 +307,7 @@ func (b *blockManager) startSync(peers *list.List) {
 		// and fully validate them.  Finally, regression test mode does
 		// not support the headers-first approach so do normal block
 		// downloads when in regression test mode.
-		if b.nextCheckpoint != nil &&
-			best.Height < b.nextCheckpoint.Height &&
-			!cfg.RegressionTest && !cfg.DisableCheckpoints {
```

Comment:
I agree about `cfg.DisableCheckpoints`, but I believe there can in fact be checkpoints for `cfg.RegressionTest` because you can manually specify them via `--addcheckpoint`.  Granted it wouldn't make sense to do that since the whole point of the regression test network is generating a new chain, so it's probably wouldn't be that big of a deal, but it would mean you can't use the regression network to detect intentionally bad checkpoints.